src/IfSharp.Kernel/helpers/FSCharting.Gtk.fsx

```diff
@@ -44,141 +45,134 @@ open OxyPlot.Series
-            //actualChart.Dock <- DockStyle.None
-            //actualChart.Size <- Size(width, height)
-            //actualChart.SaveImage(ms, ImageFormat.Png)
-            //ms.ToArray()
-
-        member self.ToData(?size) =
-            let bytes = match size with Some size -> self.ToPng(size) | _ -> self.ToPng()
-            let base64 = System.Convert.ToBase64String(bytes)
-            let data = "data:image/png;base64,"+base64
-            data
-
-    type FSharp.Charting.Chart with
-
-        /// Wraps a GenericChartWithSize around the GenericChart
-        static member WithSize(x:int, y:int) =
-
-            fun (ch : #ChartTypes.GenericChart) ->
-                ch.WithSize(x, y)
+    /// Converts the GenericChart to a PNG, in order to do this, we must show a form with ChartControl on it, save the bmp, then write the png to memory
+    member self.ToPng(?size) =
+        // get the size
+        let (width, height) = if size.IsNone then (320, 240) else size.Value
+
+        // create a new ChartControl in order to get the underlying Chart
+        //let ctl = new ChartTypes.ChartControl(self)
+
+        // save
+        use ms = new MemoryStream()
+        //let plot = new OxyPlot.GtkSharp.PlotView(Model = self.Model )
+        //let bm = self.CopyAsBitmap()
+        let pngExporter = new OxyPlot.GtkSharp.PngExporter()
+        pngExporter.Width <- width
+        pngExporter.Height <- height
+        pngExporter.Background <- OxyPlot.OxyColors.White
+        // write to a temporary file
+        //let tmp = sprintf "%s.png" (System.Guid.NewGuid().ToString())
+        pngExporter.Export(self.Model, ms) //tmp, width, height, OxyPlot.OxyColors.White)
+        //let bytes = File.ReadAllBytes(tmp);
+        // write to the stream
+        //stream.Write(bytes, 0, bytes.Length);
+        // delete the temporary file
+        //File.Delete(tmp);
+
+        //Export(self.Model, ms, width, height, e, 96)
+
+        //self.Chart.SaveImage(ms, ChartImageFormat.Png |> int |> enum)
+
+        ms.Seek(0L, System.IO.SeekOrigin.Begin) |> ignore
+        ms.ToArray()
+        //Bitmap.FromStream ms
+        //let actualChart = ctl.Controls.[0] :?> System.Windows.Forms.DataVisualization.Charting.Chart
+        //actualChart.Dock <- DockStyle.None
+        //actualChart.Size <- Size(width, height)
+        //actualChart.SaveImage(ms, ImageFormat.Png)
+        //ms.ToArray()
+
+    member self.ToData(?size) =
+        let bytes = match size with Some size -> self.ToPng(size) | _ -> self.ToPng()
+        let base64 = System.Convert.ToBase64String(bytes)
+        let data = "data:image/png;base64,"+base64
+        data
+
+type FSharp.Charting.Chart with
+
+    /// Wraps a GenericChartWithSize around the GenericChart
+    static member WithSize(x:int, y:int) =
+
+        fun (ch : #ChartTypes.GenericChart) ->
+            ch.WithSize(x, y)
+
+do
+    Printers.addDisplayPrinter(fun (x:ChartTypes.GenericChart) ->
```

Comment:
Is there some way to do github codereviews ignoring whitespace changes?

If not, perhaps consider submitting maintaining the 8-space indent so the minimal diff is more apparent